### DISCUSSION TITLE
Where to find PostgreSQL table column's description

### DISCUSSION THREAD

**Discussion by JamesInform** [osm2pgsql-dev/osm2pgsql#2109] [Q&A] [answered]
I am new to osm and PostGIS. So, probably some dump questions.

I have just imported osm data into PostgreSQL.

Now e.g. there is a table called
`planet_osm_polygon`
that has many different columns.

E.g. there is a column "horse". What is it good for? Or, how can I search for rivers?

Where can I find a description of the columns and the meaning/content they store?
Is there a detailed database schema description somewhere?

Cheers,

James

**Comment by SomeoneElseOSM** [comment, marked as answer]
> E.g. there is a column "horse". What is it good for? Or, how can I search for rivers?

The columns that are created when you load data are determined by a ".style" file that tells osm2pgsql what columns to create.  An example of one of those is [here](https://github.com/gravitystorm/openstreetmap-carto/blob/master/openstreetmap-carto.style).  Depending on what instructions you were following to get to wherever you have got to, you might have different data columns.

What goes in those columns comes essentially from OpenStreetMap, so if I do a "select osm_id from planet_osm_polygon where horse = 'yes';" among the data returned is for [osm_id 23496240](https://www.openstreetmap.org/way/23496240).

Taking a step back, and considering that osm2pgsql might be only part of what you want to use to do whatever you want t do, it might be worth asking questions at https://community.openstreetmap.org/tags/c/help-and-support/7/none/osm2pgsql .

**Comment by ImreSamu** [comment]
> E.g. there is a column "horse". What is it good for? 

https://wiki.openstreetmap.org/wiki/Key%3Ahorse
* "Use this key for mapping access permission for equestrians (people riding horses) on highways/tracks/footpaths."
* https://taginfo.openstreetmap.org/keys/horse#overview
* https://taginfo.openstreetmap.org/keys/horse#values

> Or, how can I search for rivers?

check the OSM wiki
https://wiki.openstreetmap.org/wiki/Rivers

Tags
* waterway=river  https://wiki.openstreetmap.org/wiki/Tag:waterway%3Driver
* natural=water + water=river  https://wiki.openstreetmap.org/wiki/Tag:natural%3Dwater
* waterway=riverbank (use discouraged)

so you have to find the `waterway="river"` records in the database.

For the OSM data model: 
* https://wiki.openstreetmap.org/wiki/Elements
* https://wiki.openstreetmap.org/wiki/Tags
* etc ...
